From c3f0f3c91d86939e6fabf5f65c6c6fc964e6032e Mon Sep 17 00:00:00 2001
From: Frank Denis <github@pureftpd.org>
Date: Thu, 20 Jan 2022 19:54:27 +0100
Subject: [PATCH] PostgreSQL: don't escape the port number in the connection string

It's already an integer, not a string.

Reported by Artyom Davidov, thanks!

Origin: upstream, https://github.com/jedisct1/pure-ftpd/commit/c3f0f3c91d86939e6fabf5f65c6c6fc964e6032e
Bug-Ubuntu: https://launchpad.net/bugs/2048764
Last-Update: 2024-01-09
---
 src/log_pgsql.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/log_pgsql.c b/src/log_pgsql.c
index 875b71f..bb08c3b 100644
--- a/src/log_pgsql.c
+++ b/src/log_pgsql.c
@@ -278,7 +278,6 @@ static int pw_pgsql_connect(PGconn ** const id_sql_server)
     *id_sql_server = NULL;
 
     if ((escaped_server = pw_pgsql_escape_conninfo(server)) == NULL ||
-        (escaped_port = pw_pgsql_escape_conninfo(port)) == NULL ||
         (escaped_db = pw_pgsql_escape_conninfo(db)) == NULL ||
         (escaped_user = pw_pgsql_escape_conninfo(user)) == NULL ||
         (escaped_pw = pw_pgsql_escape_conninfo(pw)) == NULL) {
@@ -296,7 +295,7 @@ static int pw_pgsql_connect(PGconn ** const id_sql_server)
     }
     if (SNCHECK(snprintf(conninfo, sizeof_conninfo,
                          PGSQL_CONNECT_FMTSTRING,
-                         escaped_server, escaped_port, escaped_db,
+                         escaped_server, port, escaped_db,
                          escaped_user, escaped_pw), sizeof_conninfo)) {
         goto bye;
     }
@@ -314,7 +313,6 @@ static int pw_pgsql_connect(PGconn ** const id_sql_server)
     bye:
     free(conninfo);
     free(escaped_server);
-    free(escaped_port);
     free(escaped_db);
     free(escaped_user);
     free(escaped_pw);
